### DISCUSSION TITLE
Migration Guid from electron to Tauri (including some technical/performance details)

### DISCUSSION THREAD

**Discussion by ghost** [tauri-apps/tauri#2359] [Q&A]
I was about to comment on this discussion but then realized github doesn't allow comments - only emojis and answers! This is an extension of the request made here:
https://github.com/tauri-apps/tauri/discussions/1869

+1M - I am sizing the effort for migrating an electron app  and even basic questions arise that this discussion may provide psuedo documentation for if anyone can give me at least the broad brush strokes on:

- Will all of my use of axios HTTP calls in javascript _just work_ running in Tauri? Or do I have to re-route all calls through rust (I hope not, as that would break the platform independence of this code still running in a browser!). I'd prefer to use "bridge" pattern but it says on the "cons" section: `Some WebAPIs unavailable`. What does that mean if I'm trying to make little json content rest (get/put/post) requests on e.g. an https rest server external to the app?
- I was attracted to [Tauri for bundle sizes theoretically below 1 MB](https://www.reddit.com/r/rust/comments/g9p2om/anybody_using_tauri/fow3pmw?utm_source=share&utm_medium=web2x&context=3), but it seems highly dependent on features used, so what would a  "bridge" or "cloudbridge" (if necessary for web apis) hello world app cost on disk space (ballpark)? My electron app is currently ~90 MB, so anything close to that would defeat the purpose.
-  On electron, virtual memory gets eaten up 4GB per V8 isolate (more than some restricted environments are willing to allow, i.e. with `ulimit`), is it a fairly safe assumption to make that Tauri "bidge" or "cloudbridge" are orders of magnitude less in its virtual memory footprint than electron?

So all in all I see value in having these answers documented since, as they say, the devil is always in the details. I don't doubt a simple "Hermit" pattern is extremely light weight, but for full-featured electron apps we developers need to know what scale of resources Tauri will consume to justify the cost of migration with our teams and business partners. I think usage would go way up if people had a short list of talking points like the OP suggests on how a Tauri migration looks, both functionally and with performance, besides those folks that really just wanna migrate so they get to work with Rust - hey me included ;)

Thank you, I remain extremely optimistic and hope to get answers soon as I am bloat-sick with electron currently. If it helps, I'd definitely buy your book if I can make it happen with this migration :)

**Comment by nothingismagick** [comment]
Hey there, your Axios calls will continue to work assuming you ship that library to your front end. You will NOT need to retool. If you need to sidestep CORS, you can route those requests through the Tauri API. The security risks here (processing responses in the webview) are known to you, obviously. If you own / run the remote endpoints, you may have to fiddle with your CSP though.

Our current bundles with full API tend toward 4MB, your mileage may vary - especially if you are not careful with your CSS or are using heavy images like JPG etc.

Our memory consumption versus Electron is being tracked here (but you can see there is generally a good deal LESS memory in Tauri apps):

https://tauri.studio/en/benchmarks

Thankyou for your concern about the general "quality" of the documentation and interest in the book. We have opted to finish the code before spending TOO much time on the docs / tutorials. Rest assured, this will be happening over the next two months though as we press on to the stable release.

**Reply by ghost** [reply]
wow, fast response thanks! And only 4MB for a full "cloudbridge" - sounds great! So there's still a part to my original question I'm curious to know:

>Some WebAPIs unavailable

For the "bridge" pattern I hope to use, what does statement mean specifically under the "cons" section of the docs? And what you said about CORS, I am familiar with chromes implementation of CORS (per electron). Anything drastically different about Tauri's webkit that's useful to know?

And one more: what use case would require "cloudbridge" instead of just "bridge". I have a lot of assets, and alias their location using webpack so that I import them simply as "assets/config.json". Would I need "cloudbridge" for that or can "bridge" support loading those assets?
